From 2d1360c28c8c11c884440578ee179c8f99f05727 Mon Sep 17 00:00:00 2001
From: Bing Sun <quic_bings@quicinc.com>
Date: Fri, 8 Jul 2022 13:44:41 +0800
Subject: [PATCH 4/4] wpa_supplicant: Make EasyMesh agent on board with hidden
 SSID

Enable scan_ssid for wps connection by default, and remove scan channel
optimization for wps. Otherwise EasyMesh agent on board fails if backhaul
BSS uses hidden SSID
---
 wpa_supplicant/scan.c           | 2 +-
 wpa_supplicant/wps_supplicant.c | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/wpa_supplicant/scan.c b/wpa_supplicant/scan.c
index 97a8d9a..996a161 100644
--- a/wpa_supplicant/scan.c
+++ b/wpa_supplicant/scan.c
@@ -446,7 +446,7 @@ static void wpa_supplicant_optimize_freqs(
 	}
 #endif /* CONFIG_P2P */
 
-#ifdef CONFIG_WPS
+#if 0
 	if (params->freqs == NULL && wpa_s->after_wps && wpa_s->wps_freq) {
 		/*
 		 * Optimize post-provisioning scan based on channel used
diff --git a/wpa_supplicant/wps_supplicant.c b/wpa_supplicant/wps_supplicant.c
index 5633f3d..81deb58 100644
--- a/wpa_supplicant/wps_supplicant.c
+++ b/wpa_supplicant/wps_supplicant.c
@@ -477,6 +477,7 @@ static int wpa_supplicant_wps_cred(void *ctx,
 
 	wpa_config_set_network_defaults(ssid);
 	ssid->wps_run = wpa_s->wps_run;
+	ssid->scan_ssid=1;
 
 	os_free(ssid->ssid);
 	ssid->ssid = os_malloc(cred->ssid_len);
-- 
2.7.4

